### PR TITLE
Pass the original mask tensor when using ds-inference at BLOOM architecture

### PR DESCRIPTION
**Pull request by RezaYazdaniAminabadi**
# What does this PR do?

This resolves DeepSpeed-Inference integration for the BLOOM model. There is no need to convert `attention_mask` to `casual_mask` when using DeepSpeed to run inference. There is [this PR ](https://github.com/microsoft/DeepSpeed/pull/2217)on the DeepSpeed side that will add `ds_inference` attribute to the transformer network, and the mask remains intact in that case. Note that this PR will not change the flow on the modeling side and will only use a different mask when one calls `deepspeed.init_inference` on the created model.
cc: @stas00 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by stas00** [comment]
cc: @thomasw21 

and also @LysandreJik - I hope it's OK if we sneak in a bit of `ds-inference` code into the BLOOM modeling code.

Unless of course you can think of how to make this configurable.

Thanks.

**Comment by thomasw21** [comment]
Hum I'm not too familiar with how DS works but would something like `self._prepare_attn_mask = lambda attention_mask, *args, **kwargs: attention_mask` work? This would live inside DS instead of the modeling?

**Comment by stas00** [comment]
That's probably a sufficient hack. 

Reza, will this work for your needs?

**Comment by RezaYazdaniAminabadi** [comment]
@stas00 Will try, if it works, I am happy to use this instead. 
Thanks @thomasw21

**Comment by RezaYazdaniAminabadi** [comment]
Okay, I have made this change on the DeepSpeed side [(here)](https://github.com/microsoft/DeepSpeed/pull/2217/commits/82a37d6d4393d9ed76a80dcb392c51e48188fd14#diff-fdbdbba9c01d8349399cfadeba4cce1c5a34fc15c38ec44a335e1340c11b13a4R177) as suggested. So, I am gonna close this PR as it not needed anymore.
Thanks,
Reza
